### PR TITLE
Add api calls productdetails

### PR DESCRIPTION
**Pull request by alizehrehman**
- Added API calls to top level ProductDetails Component
- Added state elements to top level Product Details Component
- Sent API data down to child components (Default View, ProductInfo, StyleSelector, AddToBag)
- Default View: Added render of default photo of selectedStyle
- ProductInfo: Added render of product category, title, price, style name
- StyleSelector: Added render of style icon thumbnails
- AddToBag: Added render and state management of size selection dropdown & quantity selection dropdown
- CSS: added basic css to format as CSS grid, added css to icon thumbnails for round image formatting, added basic borders to delineate component spaces
- **Config Files:** Added configuration files to help combat bug & removed babel config file (unneeded with the options listed in the webpack config) -- please incorporate into master as much as possible
Note: Things are a bit messy and might need refactor into further subcomponents to make it easier to read.

### CODE REVIEW

**Comment by alizehrehman** [comment]
@matthewwrobel @jabrake Can you please review my commit? It's a big one so I apologize! I also had to make some updates to config files while debugging, can we incorporate the updates where it makes sense? Thank you!!

File: client/product_details/AddToBag.jsx [hunk 1]
```diff
@@ -3,20 +3,79 @@ import React from 'react';
...
+        <select name={this.state.selectedSize} id="selectSize" placeholder="SELECT SIZE" onChange={this.handleSizeSubmit}>
+          <option value="SELECT SIZE">SELECT SIZE</option>
+          {this.props.availableSizes.map((size) => {
+            return (
+              <option value={size[0]}>{size[0]}</option>
+            );
+          })}
+        </select>
+        <select name={this.state.selectedQuantity} id="selectQuantity" placeholder={this.state.selectedQuantity} onChange={this.handleQuantitySubmit}>
+          {quantityArr.map((quantity) => {
```
Comment (matthewwrobel) [inline client/product_details/AddToBag.jsx]: tiny detail, but you could get rid of the curly braces and return statement and do this on one line I think--line 60, too.  Everything is looking awesome so far.

File: client/product_details/AddToBag.jsx [hunk 2]
```diff
@@ -3,20 +3,79 @@ import React from 'react';
...
+  handleQuantitySubmit(e) {
+    var selectedQuantity = e.target.value;
+    this.setState({
+      selectedQuantity: selectedQuantity
+    });
   }
 
   render() {
+    var quantityArr = [];
+    for (var i = 1; i < (this.state.maxQuantity + 1); i++) {
```
Comment (matthewwrobel) [inline client/product_details/AddToBag.jsx:51]: nice!

File: client/product_details/DefaultView.jsx [hunk 3]
```diff
@@ -3,16 +3,27 @@ import React from 'react';
...
+    if (this.props.selectedStyle !== undefined) {
+      var photoGallery = this.props.selectedStyle.photos;
+      return (
+        <div id="defaultView">
+          <a target="_blank" href={photoGallery[this.state.currPhotoIndex].url}>
+            <img src={photoGallery[this.state.currPhotoIndex].url} alt={this.props.selectedStyle.name} className="defaultView" id={this.props.selectedStyle.style_id}></img>
+          </a>
+        </div>
+      );
+    } else {
```
Comment (matthewwrobel) [inline client/product_details/DefaultView.jsx:22]: I don't think you need the else here becuase of the return statement in the above if block. But that' s also nit-picky. Everything is looking really great.

File: client/product_details/ProductDetails.jsx [hunk 4]
```diff
@@ -11,9 +11,44 @@ import ProductOverview from './ProductOverview.jsx';
 class ProductDetails extends React.Component {
   constructor(props) {
     super(props);
+    
+    fetch(`http://localhost:8080/products/${this.props.product_id}`)
+      .then((response) => {
+        return response.json();
+      }) 
+      .then((data) => {
+        this.setState({
```
Comment (matthewwrobel) [inline client/product_details/ProductDetails.jsx:21]: You can move this into the above .then() block right after response.json() (which is synchronous) to cut out this .then()

File: client/product_details/ProductDetails.jsx [hunk 5]
```diff
@@ -11,9 +11,44 @@ import ProductOverview from './ProductOverview.jsx';
...
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+    
+    fetch(`http://localhost:8080/products/${this.props.product_id}/styles`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
```
Comment (matthewwrobel) [inline client/product_details/ProductDetails.jsx:33]: same as above

File: client/product_details/ProductDetails.jsx [hunk 6]
```diff
@@ -11,9 +11,44 @@ import ProductOverview from './ProductOverview.jsx';
...
+      });
+
     this.state = {
       view: 'default',
-      styleSelected: ''
+      selectedStyle: undefined,
+      info: undefined,
+      styleInfo: undefined,
     };
   }
```
Comment (matthewwrobel) [inline client/product_details/ProductDetails.jsx:55]: Def curious to talk about fetching from the constructor. I have it the same way as you do with .setState called in the .then() blocks. Leslie told me that wasn't how to do it and I'm confused!

**Review by matthewwrobel — APPROVED** [review]
This looks great. I'm gonna trust you on all the webpack/babel business :). Also I suck at CSS so I didn't look at that very closely but I'm sure its great.

**Comment by alizehrehman** [comment]
Made a few updates based on PR comments. Ready to merge!

**Pull request merged**